pkg/diagnose/ports/ports.go

```diff
@@ -0,0 +1,91 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package ports provides a diagnose suite for the ports used in the agent configuration
+package ports
+
+import (
+	"fmt"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
+	"github.com/DataDog/datadog-agent/pkg/util/port"
+)
+
+var agentNames = map[string]struct{}{
+	"datadog-agent": {}, "agent": {}, "trace-agent": {},
+	"process-agent": {}, "system-probe": {}, "security-agent": {},
+	"dogstatsd": {},
+}
+
+// DiagnosePortSuite displays information about the ports used in the agent configuration
+func DiagnosePortSuite() []diagnosis.Diagnosis {
+	ports, err := port.GetUsedPorts()
+	if err != nil {
+		return []diagnosis.Diagnosis{{
+			Name:      "ports",
+			Result:    diagnosis.DiagnosisUnexpectedError,
+			Diagnosis: fmt.Sprintf("Unable to get the list of used ports: %v", err),
+		}}
+	}
+
+	portMap := make(map[uint16]port.Port)
+	for _, port := range ports {
+		portMap[port.Port] = port
+	}
+
+	var diagnoses []diagnosis.Diagnosis
+	for _, key := range config.Datadog.AllKeysLowercased() {
+		splitKey := strings.Split(key, ".")
+		keyName := splitKey[len(splitKey)-1]
+		r, _ := regexp.Compile("_?port_?")
+		if !r.MatchString(keyName) {
+			continue
+		}
```

Comment:
This would match any string containing port, not necessarily with _ before or after... (eg. `export`)
Was there any issue with the previous version ?
```suggestion
		if keyName != "port" && !strings.HasPrefix(keyName, "port_") && !strings.HasSuffix(keyName, "_port") {
			continue
		}
```

Reply:
Ah, I was trying to change to regex based on Gustavo's previous suggestion, but the regex pattern he used was too restrictive. I think for now we can switch back to this.